miso-web/src/main/webapp/scripts/library_hot.js

```diff
@@ -8,12 +8,16 @@ Library.hot = {
   libraryTypeAliases: {},
   
   getLibraryTypeAliasLists: function () {
-    Hot.dropdownRef.libraryTypes.forEach(function (lt) {
-      if (Library.hot.libraryTypeAliases[lt.platform]) {
-        Library.hot.libraryTypeAliases[lt.platform].push(lt.alias);
-      } else {
-        Library.hot.libraryTypeAliases[lt.platform] = [lt.alias];
+    var usedTypes = Library.hot.librariesJSON.map(function (lib) {
+      return lib.libraryTypeId;
+    });
+    Hot.dropdownRef.libraryTypes.filter(function (lt) {
+      return !lt.archived || usedType.indexOf(lt.id) != -1;
+    }).forEach(function (lt) {
+      if (!Library.hot.libraryTypeAliases[lt.platform]) {
+        Library.hot.libraryTypeAliases[lt.platform] = [];
       }
+      Library.hot.libraryTypeAliases[lt.platform].push(lt.alias);
```

Comment:
But if you don't change the platform then it doesn't set it, so if the user is editing a library and tries to change the libraryType without changing the platform then they don't see any libraryType dropdown options.